var/spack/repos/builtin/packages/mpifileutils/package.py

```diff
@@ -39,18 +42,74 @@ class Mpifileutils(AutotoolsPackage):
 
     depends_on('libarchive')
 
+    depends_on('cmake@3.1', when='@0.9:', type='build')
+
+    variant('build_type', default='RelWithDebInfo',
```

Comment:
Doesn't this variant come from CMakePackage? Why does it need to be created here again?

Reply:
Packages inheriting from `DynamicInheritancePackage` will have to shadow the directives from the parent class, since they need access to those before they get a chance to set their inheritance. That's variants, dependencies, and conflicts. If anyone has a brilliant idea let me know, but as of now I don't know of any way to do it other than shadowing those.